### ISSUE TITLE
Error:use tensorflow_cifar-10 model on tensorflow android camera

### ISSUE THREAD

**Issue by qemb01** [tensorflow/tensorflow#2923]
I am very new to tensorflow. Now I am trying to use tensorflow_cifar-10 model on tensorflow android camera, the app is running but there is no output.

At first I have tried this example:
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/android
This example uses the .pb file at:
https://storage.googleapis.com/download.tensorflow.org/models/inception5h.zip
It works well.

Then I tried to replace the .pb file with the tensorflow_cifar-10. I followed some of the instructions here:
https://github.com/tensorflow/tensorflow/issues/616
and add the save__graph() function in /tensorflow/models/image/cifar10/cifar10_train.py

code:

def save_graph():

    g = tf.Graph()
    vars = {}
    with g.as_default():
        with tf.Session() as sess:
            d = np.ones([1,24,24,3],dtype=np.float32)
            input_data = tf.placeholder(tf.float32,shape=[1,24,24,3], name="input_placeholder")
            images, labels = cifar10.distorted_inputs()
            logits = cifar10.inference(images)
            init = tf.initialize_all_variables()
            sess.run(init)
            saver = tf.train.Saver(tf.trainable_variables(),max_to_keep=0)
            saver.restore(sess,os.path.join(FLAGS.train_dir, 'model.ckpt'))
            print (sess.run(logits,{images:d}))
            for v in tf.trainable_variables():
                vars[v.value().name] = sess.run(v)

    g2 = tf.Graph()
    consts = {}
    with g2.as_default():
        with tf.Session() as sess:
            for k in vars.keys():
                consts[k] = tf.constant(vars[k])
            tf.import_graph_def(g.as_graph_def(),input_map={name:consts[name] for name in consts.keys()})
            tf.train.write_graph(sess.graph_def,'/home/eli/Documents/TensorflowProjetos/ProtobufFiles','graph.pb',False)

    return os.path.join('/home/eli/Documents/TensorflowProjetos/ProtobufFiles','graph.pb')

I use the save__graph() to get the .pb file after the end of training. Then I replace the tensorflow_inception_graph.pb, whice is from inception5h.zip and used in tensorflow_android_camera, with the new .pb file from the cifar10 model.

Finally I run the tensorflow_android_camera app but there is no output. The error in log is: 
Error during inference: Not found: FeedInputs: unable to find feed output input:0
I do not know why it happened, please help me ,thank you!

**Comment by sherrym** [comment]
You will need to either name your input_data "input", or change the demo code to feed "input_placeholder:0".

**Comment by qemb01** [comment]
@sherrym I have done this, but nothing is changed.

**Comment by prb12** [comment]
@qemb01 Please can you supply the exact code you are now running and the command lines necessary to reproduce this problem?

**Comment by qemb01** [comment]
@prb12 Of course, here is my code in cifar10_train.py:

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

from datetime import datetime
import os.path
import time
import os

import numpy as np
from six.moves import xrange  # pylint: disable=redefined-builtin
import tensorflow as tf

import os
from tensorflow.python.platform import gfile

from tensorflow.models.image.cifar10 import cifar10

FLAGS = tf.app.flags.FLAGS

tf.app.flags.DEFINE_string('train_dir', '/tmp/cifar10_train',
                           """Directory where to write event logs """
                           """and checkpoint.""")
tf.app.flags.DEFINE_integer('max_steps', 20000,
                            """Number of batches to run.""")
tf.app.flags.DEFINE_boolean('log_device_placement', False,
                            """Whether to log device placement.""")


def train():
  """Train CIFAR-10 for a number of steps."""
  with tf.Graph().as_default():
    global_step = tf.Variable(0, trainable=False)

    # Get images and labels for CIFAR-10.
    images, labels = cifar10.distorted_inputs()

    # Build a Graph that computes the logits predictions from the
    # inference model.
    logits = cifar10.inference(images)

    # Calculate loss.
    loss = cifar10.loss(logits, labels)

    # Build a Graph that trains the model with one batch of examples and
    # updates the model parameters.
    train_op = cifar10.train(loss, global_step)

    # Create a saver.
    saver = tf.train.Saver(tf.all_variables())

    # Build the summary operation based on the TF collection of Summaries.
    summary_op = tf.merge_all_summaries()

    # Build an initialization operation to run below.
    init = tf.initialize_all_variables()

    # Start running operations on the Graph.
    sess = tf.Session(config=tf.ConfigProto(
        log_device_placement=FLAGS.log_device_placement))
    sess.run(init)


    # Start the queue runners.
    tf.train.start_queue_runners(sess=sess)

    summary_writer = tf.train.SummaryWriter(FLAGS.train_dir,
                                            graph_def=sess.graph_def)

    for step in xrange(FLAGS.max_steps):
      start_time = time.time()
      _, loss_value = sess.run([train_op, loss])
      duration = time.time() - start_time

      assert not np.isnan(loss_value), 'Model diverged with loss = NaN'

      if step % 10 == 0:
        num_examples_per_step = FLAGS.batch_size
        examples_per_sec = num_examples_per_step / duration
        sec_per_batch = float(duration)

        format_str = ('%s: step %d, loss = %.2f (%.1f examples/sec; %.3f '
                      'sec/batch)')
        print (format_str % (datetime.now(), step, loss_value,
                             examples_per_sec, sec_per_batch))

      if step % 100 == 0:
        summary_str = sess.run(summary_op)
        summary_writer.add_summary(summary_str, step)

      # Save the model checkpoint periodically.
      if step % 1000 == 0 or (step + 1) == FLAGS.max_steps:
        checkpoint_path = os.path.join(FLAGS.train_dir, 'model.ckpt')
        saver.save(sess, checkpoint_path, global_step=step)

def save_graph():

    g = tf.Graph()
    vars = {}
    with g.as_default():
        with tf.Session() as sess:
            d = np.ones([128,24,24,3],dtype=np.float32)

#            input_data = tf.placeholder(tf.float32,shape=[128,24,24,3], name="input")
            images, labels = cifar10.distorted_inputs()
            logits = cifar10.inference(images)

            init = tf.initialize_all_variables()
            sess.run(init)
            saver = tf.train.Saver(tf.trainable_variables(),max_to_keep=0)
            saver.restore(sess,os.path.join(FLAGS.train_dir, 'model.ckpt-19999'))

            print (sess.run(logits,{images:d}))
            for v in tf.trainable_variables():
                vars[v.value().name] = sess.run(v)


    g2 = tf.Graph()
    consts = {}
    with g2.as_default():
        with tf.Session() as sess:
            for k in vars.keys():
                consts[k] = tf.constant(vars[k])
            tf.import_graph_def(g.as_graph_def(),input_map={name:consts[name] for name in consts.keys()})

            tf.train.write_graph(sess.graph_def,'/home/eli/Documents/TensorflowProjetos/ProtobufFiles','graph.pb',False)

    return os.path.join('/home/eli/Documents/TensorflowProjetos/ProtobufFiles','graph.pb')


def main(argv=None):  # pylint: disable=unused-argument
  cifar10.maybe_download_and_extract()
  if tf.gfile.Exists(FLAGS.train_dir):
    tf.gfile.DeleteRecursively(FLAGS.train_dir)
  tf.gfile.MakeDirs(FLAGS.train_dir)
  train()
  save_graph()

if __name__ == '__main__':
  tf.app.run()


I have run the cifar10_train.py to get pb file. Then I replace the tensorflow_inception_graph.pb with it. The tensorflow_inception_graph.pb is in the /tensorflow/tensorflow/tree/master/tensorflow/examples/android/assets/ of tensorflow android camera demo.

Tensorflow android camera demo:
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/android

**Comment by prb12** [comment]
I'm sorry - but this code looks pretty confused to me.  As @sherrym said above, the android example is expecting a graphdef with a placeholder node called 'input' into which it will presumably feed images from the camera.  Your _original_ code had an *unused* placeholder node called input_data: 
```
input_data = tf.placeholder(tf.float32,shape=[1,24,24,3], name="input_placeholder")
```
The _new_ code above calls cifar10.distorted_inputs() which I believe generates a graph to read input images from files in the filesystem (whose names are retrieved from a queue!) and randomly mutates them.  If you want this to work you will need to create a graphdef which accepts images via a node called 'input' and feeds them into cifar10.inference. 

Since it isn't a TensorFlow bug, but really a general programming question, you should probably re-ask this on StackOverflow. (please ensure you tag the question 'tensorflow').

**Comment by qemb01** [comment]
@prb12 Thank you very much. I will try to do it.

**Comment by codeloverr** [comment]
@qemb01  Have you successful to replace the model file?
